### PR TITLE
Sourcegraph 3.0 announcement

### PR DESCRIPTION
**Pull request by sqs**
- Major difference from the 3.0 beta announcement is how basic code intel is described.
- Removes most text from the 3.0 beta announcement to avoid confusion.

closes https://github.com/sourcegraph/sourcegraph/issues/1959

### CODE REVIEW

File: blogposts/announcing-sourcegraph-3.0.md [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+- [Sourcegraph extensions](https://docs.sourcegraph.com/extensions) are the sole source of code intelligence now. We removed the old way, which required a lot more Sourcegraph-specific custom integration work to add or configure a new language.
+- The [management console](https://docs.sourcegraph.com/admin/management_console) makes it so that all configuration can be edited online and in-band, which means future Sourcegraph releases can auto-migrate their own config instead of requiring manual (and error-prone) steps by the site admin.
+- The new concept of [external service connections](https://docs.sourcegraph.com/admin/repo/add) will allow us to improve Sourcegraph's synchronization of repositories in a future release.
+- Standardizing on Nginx for serving HTTP lets us remove a lot of complex core code to support numerous use cases better handled by Nginx.
+
+## Thank you
+
+Thank you to the many people who contributed to make Sourcegraph better since our last release!
+
+- [@sabyasachi](https://github.com/sabyasachi)
```
Comment (sqs) [inline blogposts/announcing-sourcegraph-3.0.md]: @ryan-blunden 	Could you please do your magic and update this list to add contributors between 3.0 beta and now?

File: blogposts/announcing-sourcegraph-3.0.md [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+
+We want anyone to be able to start using Sourcegraph for code search without reading a manual. That's why we made Sourcegraph search super fast and added one-click query suggestions (computed locally using heuristics) to refine your query.
+
+But when you're searching more than a couple times daily, learning Sourcegraph's search keywords (such as `repo:`) will make you more productive. Click the new <kbd>?</kbd> icon next to any search button to see a quick reference. See ["Search query syntax"](https://docs.sourcegraph.com/user/search/queries) for more info.
+
+[Try it out](https://sourcegraph.com/search?q=repo:%5Egithub.com/kubernetes/kubernetes%24+lang:go+-f:test+-f:vendor+controller) (click the <kbd>?</kbd> next to the search button):
+
+[![Search query cheatsheet](./announcing-sourcegraph-3.0/search_query_cheatsheet.png)](https://sourcegraph.com/search?q=repo:%5Egithub.com/kubernetes/kubernetes%24+lang:go+-f:test+-f:vendor+controller)
+
+### Code intelligence for all languages
```
Comment (sqs) [inline blogposts/announcing-sourcegraph-3.0.md]: @chrismwendt FYI here is how I am describing the basic code intel improvements. I think the copy could be a bit better, but it's a start.

cc @ryan-blunden

File: blogposts/announcing-sourcegraph-3.0.md [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+
+It works immediately for all code with zero configuration because it uses heuristics to provide fuzzy results by default. This is surprisingly helpful and fast, and it finds results in your other repositories, too.
+
+Want precise results? Many of the [N language extensions](https://sourcegraph.com/extensions?query=category%3A%22Programming+languages%22) also can provide precise results, by building and analyzing the code like a compiler, often by using a language server. See each extension's page (such as the [TypeScript/JavaScript extension](https://sourcegraph.com/extensions/sourcegraph/typescript)) for details.
+
+As long as you're aware of the limitations of the default (fuzzy) mode, you'll like it lot. For example, in the fuzzy mode, if multiple functions in your code are named `open`, the reference results for `open` will include false positives. (If it bothers you, toggle it off temporarily in the command palette or disable the language extension completely.)
```
Comment (chrismwendt) [inline blogposts/announcing-sourcegraph-3.0.md]: Once https://github.com/sourcegraph/sourcegraph/issues/2035 is complete, we won't have the concept of fuzzy vs precise anymore, right? It looks like the toggle for fuzzy mode enables/disables other language extensions https://github.com/sourcegraph/sourcegraph-basic-code-intel/blob/c25813e581f1e94b26eee1cff2e2b2615f91da7f/src/extension.ts#L41 but that won't be necessary once we have individual language extensions for all the popular languages.

File: blogposts/announcing-sourcegraph-3.0.md [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+
+Also, in case you missed it, [Sourcegraph is now open source](https://about.sourcegraph.com/blog/sourcegraph-is-now-open-source) (Apache-licensed).
+
+Want to hear about new features and releases as soon as they're available? Follow [@srcgraph](https://twitter.com/srcgraph).
+
+Ready to [install or upgrade a self-hosted Sourcegraph instance](#install-or-upgrade)? You can also [try these features on Sourcegraph.com for open-source code](https://sourcegraph.com).
+
+## Highlights
+
+### Search query cheatsheet
```
Comment (sqs) [inline blogposts/announcing-sourcegraph-3.0.md]: Need to move this section down 1 to match TOC

File: blogposts/announcing-sourcegraph-3.0.md (hunk 1 above)
Comment (ryan-blunden) [reply blogposts/announcing-sourcegraph-3.0.md]: On it!

File: blogposts/announcing-sourcegraph-3.0.md (hunk 3 above)
Comment (sqs) [reply blogposts/announcing-sourcegraph-3.0.md]: - The Fuzzy/Precise toggle button in the file header is only for 2.13 and earlier. 3.0 beta and newer do not show that button; instead they show the "Show/hide fuzzy results" in the panel header.
- Once https://github.com/sourcegraph/sourcegraph/issues/2035 is complete, we will still have these concepts, but they will not be exposed to the user. Fuzzy means "no lang server is configured", precise means "a lang server is configured".

Is that what you were thinking too?

Comment (chrismwendt) [reply blogposts/announcing-sourcegraph-3.0.md]: Hmm, I still can't think of a reason to display a "Show/hide fuzzy results" button.

Here's how I imagine the UX once basic-code-intel is replaced by individual language extensions:

- For languages that don't have language servers yet (everything except Go, TypeScript, and Python), the user will get code intel powered by ctags + search
- For languages that do:
  - If the user has not specified `serverUrl` yet, they will get code intel powered by ctags + search
  - If they have, they will get code intel powered by the language server

Does that match your expectation of the end result of this task of splitting basic-code-intel into individual extensions?

Comment (sqs) [reply blogposts/announcing-sourcegraph-3.0.md]: Yeah, we are good. I didn’t intend to imply that the show/hide-fuzzy button would continue to be shown.

There are cases when you want fuzzy results while you’re waiting for precise results. Imagine browsing a big TS or Go repo and you just want to find references on a fairly unique name. We can handle that in 3.1 though.

Comment (sqs) [reply blogposts/announcing-sourcegraph-3.0.md]: Updated this wording

**Comment by sqs** [comment]
This is ready to publish, but it will wait for the PostgreSQL-related work so that 3.0 beta instances have a way to preserve their data when upgrading to 3.0 (https://sourcegraph.slack.com/archives/C07KZF47K/p1549222364688200?thread_ts=1549206868.676300&cid=C07KZF47K).

**Pull request merged**